pkg/telemetry/stats_telemetry.go

```diff
@@ -11,6 +11,7 @@ import "sync"
 type StatsTelemetrySender interface {
 	Count(metric string, value float64, hostname string, tags []string)
 	Gauge(metric string, value float64, hostname string, tags []string)
+	GaugeNoIndex(metric string, value float64, hostname string, tags []string)
```

Comment:
While this works, this feels unfortunate that we would need to eventually duplicate all methods for what is otherwise just a boolean flag. Would it make  sense to add it as a parameter? Or maybe a flag to the provider that would set it on all metrics submitted with it?

Reply:
A metric not indexed is very rare. I think it makes sense to provide a way to send a gauge without having to add `false` most of the time. I don't expect there is a need soon for `CountNoIndex` for example.
As `checkSender.Gauge` is called at 98 locations, having both  `checkSender.Gauge` and `checkSender.GaugeNoIndex` makes also sense.

I think you raise a valid point because `StatsTelemetrySender ` shouldn't use directly `checkSender` and instead should share a common implementation (Making this change is out of the scope of this PR). The current design leads having 2 `GaugeNoIndex` methods.
What is your point of view?